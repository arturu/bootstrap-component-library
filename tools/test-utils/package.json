{
  "name": "@openeuropa/bcl-test-utils",
  "version": "0.1.3",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
