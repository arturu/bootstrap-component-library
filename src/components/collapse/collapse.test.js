import { merge, renderTwigFileAsNode } from "@oe/test-utils";
import demoData from "@oe/data-collapse/data";

const template = "@oe/collapse/collapse.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Collapse", () => {
  test(`renders with button trigger and default text`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData})
    ).resolves.toMatchSnapshot();
  });
});
