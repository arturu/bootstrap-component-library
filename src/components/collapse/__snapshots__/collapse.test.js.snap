// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Collapse renders with demoData 1`] = `
<button
  aria-controls="collapseExample"
  aria-expanded="false"
  autocomplete="off"
  class="btn btn-danger btn-md"
  data-bs-target="#collapseExample"
  data-bs-toggle="collapse"
  type="button"
>
  I am a button
</button>
`;
